### ISSUE TITLE
Different extended type conversion rules for static and extension calls

### ISSUE THREAD

**Issue by auduchinok** [dotnet/fsharp#13507]
```csharp
public static class Class
{
    public static void Method(this System.IEnumerable<object> s)
    {
    }
}
```

```fsharp
Class.Method([1])
[1].Method()
```

<img width="318" alt="Screenshot 2022-07-13 at 19 39 58" src="https://user-images.githubusercontent.com/3923587/178796508-f0ebb715-f786-451e-b5b2-de05edf91de2.png">

**Comment by dsyme** [comment]
Yes, this is expected - in one the element type "obj" flows into the inference for the list, in the other it doesn't.

**Comment by auduchinok** [comment]
@dsyme Is there a change this could be changed in a future F# version? Could we have a bit of special logic for this kind of extension method invocation? Or flowing of the types from right to left would be too difficult to get right here?

**Comment by dsyme** [comment]
Please go via fslang-suggestions to discuss this, since there are many ramifications to any change in this area. Thanks :)

**Comment by auduchinok** [comment]
Sure, will move there. I was trying to estimate if I should be filing an issue there or not :)
